src/handlers/gestures/GestureDetector.tsx

```diff
@@ -416,6 +417,28 @@ export const GestureDetector: React.FunctionComponent<GestureDetectorProps> = (
     );
   }
 
+  // @ts-ignore @typescript-eslint/ban-ts-comment
+  if (process.env.JEST_WORKER_ID) {
+    for (const handler of gesture) {
+      const handlers = handler.handlers;
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      const reaGestureHandler = Reanimated.useAnimatedGestureHandler({
+        onStart: handlers.onBegin,
+        onActive: handlers.onUpdate,
+        onEnd: handlers.onEnd,
+        onFinish: handlers.onFinalize,
```

Comment:
```suggestion
        // this one passes the context to the onBegin callback, but I think it's fine in tests
        onStart: handlers.onBegin,
        onActive: (event, context) => {
          // useAnimatedGestureHandler passes all events with state === 4 (ACTIVE) to the onActive
          // callback, event the StateChangeEvent from 2 (BEGAN) to 4. Since the StateChangeEvent
          // will always be the first in the new state we can just check a flag here.
          if (context.active) {
            handlers.onUpdate?.(event);
          } else {
            handlers.onStart?.(event);
            context.active = true;
          }
        },
        onEnd: (event, context) => {
          // in the new API onEnd, onFail and onCancel callbacks are merged into one, because
          // - from the user's perspective there is no meaningfull difference between gesture failing
          //   and being cancelled, so having separate callbacks for it makes no sense
          // - most of the time you want to clean up after the gesture in all of those callbacks,
          //   so there is no point in repeating the same logic three times
          // new onEnd callback receives a bool as the second argument which tells whether the gesture
          // has finished gracefully or not, and if you really need to know if it failed or was cancelled
          // you can chech the state property of event
          handlers.onEnd?.(event, true);
          context.success = true;
        },
        onCancel: (event, context) => {
          handlers.onEnd?.(event, false);
          context.success = false;
        },
        onFail: (event, context) => {
          handlers.onEnd?.(event, false);
          context.success = false;
        },
        onFinish: (event, context) => {
          // onFinalize, like onEnd, requires a flag that tells whether the gesture finished gracefully
          handlers.onFinalize?.(event, context.success ?? false);
        },
```

Reply:
I'm not sure if emulating the new API using the old one is the best approach.